### PR TITLE
Update to netstandard2.0

### PR DESCRIPTION
**Pull request by gordrs**
Hey folks,

We've moved all of our stuff to dotnet core 3.1 and are going cross platform, but Prime is still on framework. This pull request fixes that by moving Prime and Prime.Scripting to netstandard2.0 and Prime.Tests to .NET Core 3.1

I'm not an expert in xunit so I can't get the tests to show, so willing to work with someone to have that last piece sorted.

Both Prime and Prime.Scripting build with zero warnings.

### CODE REVIEW

**Comment by bryanedds** [comment]
Hi!

Thanks for this!

Please correct me if I'm wrong, but don't .NET Framework dlls work fine in .NET Core applications? From my understanding, .NET Core will have no trouble consuming .NET Framework dlls so long as those dlls do not call out to APIs that .NET Core does not support (which Prime does not).

Other than getting rid of an optional warning on the client application, I'm wondering what else this PR buys us?

Thanks again!

**Comment by gordrs** [comment]
Hi Bryan, thanks for getting back to me.

> Please correct me if I'm wrong, but don't .NET Framework dlls work fine in .NET Core applications? From my understanding, .NET Core will have no trouble consuming .NET Framework dlls so long as those dlls do not call out to APIs that .NET Core does not support (which Prime does not).

They only work fine if they're compiled to .NET Standard that allows them to be used by both Framework and Core. Core is the basis for .NET 5 so this change has to take place at some point. In this instance for this PR, I've set it as netstandard2.0.

In our use case, over the last six months we've ported our massive codebase, a large chunk of that is F# for financial applications over to .NET Core 3.1 and we make heavy use of Prime in there. In doing so, we're able to run our services on many more platforms. One of our core libraries that uses Prime is now also being used for additional cross platform uses, in particular for iOS and Android in Xamarin applications which has prompted my PR.

In short, it buys you additional supported platforms and prepares for .NET 5 in November.

Regarding the tests, I'd be more than happy to put them back to .NET Framework for now since it seems I'm not the only one having issues with F# tests on core at the moment that was only made apparent to me after I submitted the PR.

**Comment by bryanedds** [comment]
Hi again!

Thanks for giving me more details on your use-case!

I will figure out the best way to integrate this PR over the next day or two.

**Comment by gordrs** [comment]
Hi Bryan,

I've updated the project to get the tests to work under .NET Core 3.1. All is well and all tests pass.

Thanks.

**Comment by bryanedds** [comment]
Hi again!

I wanted to figure out what error was being caused by using a .NET Framework dll from a .NET Core 3.1 project, so I created an empty project, added Prime and Prime.Scripting as dependencies, invoked some simple code from each, built and ran. However, I did not experience any errors.

Can you tell me what errors you're experiencing with .NET Core 3.1 that involved the .NET Framework version of Prime? Maybe they're unrelated to this.

Thanks!

**Comment by gordrs** [comment]
Versions unrelated yes, but had to move them to netstandard2.0 in order to use them on Xamarin.

**Comment by bryanedds** [comment]
Can you tell me what the specific issue is with Xamarin? Is that on mono? Sorry, I'm a bit confused.

**Comment by gordrs** [comment]
Framework libraries do not work on Xamarin. For maximum compatability across all the platforms until they're unified in .NET 5, one has to build them to the .NET Standard. https://docs.microsoft.com/en-us/dotnet/standard/net-standard. Building libraries to the .NET Standard allows them to have the maximum cross platform usage without any problems using them. For example when I tried to bring Prime into a Xamarin application, it won't work because it only targets .NET Framework.

Yes, Xamarin is based on the mono runtime, and in this case, netstandard2.0 allows them to be used.

This is work that would have had to have taken place with the release of .NET 5 that comes out in November if Prime intended to support future versions of .NET. Framework and Core are dead as of November though Microsoft will still provide support for a number of years, but this is where all new development is going. Getting Prime to build as netstandard2.0 is the first step in that.

I've already packaged this up in a private nuget repository as netstandard2.0 and is working fine, it lets me bring in and use Prime on Xamarin. My hope is that I can eliminate that private repository if these changes are agreed upon and merged.

**Comment by bryanedds** [comment]
I'm going to merge your PR as a branch then figure out what to do about merging it into master. My only reservation hinges on whether a .NET Standard dll can be used from a .NET Framework application like my Nu Game Engine on windows as well as on Mono. The Mono 'port' of Nu was a little tricky and I have to take great care when altering any of its dependencies.

I'll let you know how things proceed!

**Comment by bryanedds** [comment]
Okay, I've got your changes plus some fixes for release builds on the https://github.com/bryanedds/Prime/tree/cryptisk-grs-master branch.

I'll have to work with another developer to ensure that this branch build will work with Nu on Mono. Let you know how that goes.

**Comment by gordrs** [comment]
Standard dlls can absolutely be used from Framework, we're using standard libraries in mono (Xamarin) core (Linux) and framework (Windows) applications. I've been doing a whole set of testing on all three of our platforms all weekend with Prime built like this and so far there's been zero issues with our tests on all platforms passing.

Thank you for taking on this PR and I'll be happy to work with you fellas to resolve any issues that crop up.

**Comment by ImaginaryDevelopment** [comment]
Is there any news on this? I'd really like Nu to be able to run on .net core and linux without mono. Really glad I checked the PRs right before I cloned the project to do this.

**Comment by bryanedds** [comment]
@ImaginaryDevelopment - ya, this still needs to be resolved as soon as I can prioritize it. However, it's not high on the priority list currently.

**Comment by gordrs** [comment]
Do you require assistance to get this up to scratch to be merged?

**Comment by bryanedds** [comment]
@cryptisk-grs - The only difficulty I think is getting time on our team to test this with the Nu Game Engine deployment on Linux w/ Mono. We've started production on a game and are focused on that right now. Prime is just the basic library of the much larger Nu Game Engine project.

However, if this is a blocking issue for your team or causing an undue maintenance burden, we can prioritize it on our side. Just let me know.

**Comment by bryanedds** [comment]
This has been resolved by other means converting Prime to netstandard2.0.
